src/Symfony/Component/DependencyInjection/Compiler/PriorityTaggedServiceTrait.php

```diff
@@ -82,6 +82,7 @@ private function findAndSortTaggedServices(string|TaggedIteratorArgument $tagNam
 
                 if (null !== $indexAttribute && isset($attribute[$indexAttribute])) {
                     $index = $attribute[$indexAttribute];
+                    $index = $container->getParameterBag()->resolveValue($index) ?: $index;
```

Comment:
why the fallback to $index if the resolved value is falsy?

Reply:
Good point. It is probably better to fall back to the defaultIndexMethod in this case.